adapters/nftkey/index.js

```diff
@@ -0,0 +1,139 @@
+        this.sdk = this.loadSdk();
+        this.symbol = s;
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Avalanche(this);
+    };
+
+    getSymbol = async () => {
+        const resp = await axios.get(
+            `${URL}/token-metadata?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyer = event => {
+        if (event.event === "TokenBidAccepted") {
+            return event.returnValues.bid.bidder;
+        }
+        return event.returnValues.buyer;
+    };
+
+    _getPrice = async (event, block) => {
+        if (!this.symbol.decimals) {
+            return { price: null, priceUsd: null };
+        }
+        
+        const po = await this.getPrice(block.timestamp);
+
+        let value = 0;
+        if (event.event === "TokenBidAccepted") {
+            value = event.returnValues.bid.value;
+        } else {
+            value = event.returnValues.listing.value;
+        }
+        const nativePrice = new BigNumber(value).dividedBy(10 ** this.symbol.decimals);
+
+        return {
+            price: nativePrice.toNumber(),
+            priceUsd: nativePrice.multipliedBy(po.price).toNumber(),
+        };
+    };
+
+    getSeller = event => {
+        if (event.event === "TokenBidAccepted") {
+            return event.returnValues.seller;
+        }
+        return event.returnValues.listing.seller;
+    };
+
+    process = async event => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const buyer = this.getBuyer(event);
+        if (!buyer) {
+            return;
+        }
+
+        const { price, priceUsd } = await this._getPrice(event, block);
+
+        const tokenId = event.returnValues.tokenId;
+        const seller = this.getSeller(event);
+        const entity = {
+            provider_name: this.name,
+            provider_contract: this.contract,
+            protocol: this.protocol,
+            nft_contract: this.contract,
+            nft_id: tokenId,
+            token: this.token,
```

Comment:
Payment token is not always `AVAX`. I've already noticed `AVAX` and `WAVAX`.
Please implement a functionality to get correct payment token.

Reply:
i've thought of this problem actually.
but from the solidity contract code, the `_paymentToken` should be immutable after construction
```  
constructor(address _paymentTokenAddress) {
        _paymentToken = IERC20(_paymentTokenAddress);
    }

    IERC20 private immutable _paymentToken;
```

Reply:
oh, i know the reason why both avax and wavax exists.
`buyToken` method will not use the `_paymentToken` field to do the payment, it will just use the native avax value.
but `acceptBidForToken` will use this field.
i will change this logic accordingly, thanks

Reply:
@liutaurasjanuskas fixed, please take a look :)